R/detect_biallelic_markers.R

```diff
@@ -79,11 +79,17 @@ detect_biallelic_markers <- function(data, verbose = FALSE) {
     }
     alt.num <- NULL
   } else {
-    # 10% of the markers are randomly sampled
-    sampled.markers <- unique(input$MARKERS)
-    sampled.markers <- sample(x = sampled.markers,
-                              size = length(sampled.markers) * 0.10,
-                              replace = FALSE)
+
+    # If there are less than 100 markers, sample all of them
+    if (length(unique(input$MARKERS)) < 100) {
+      sampled.markers <- unique(input$MARKERS)
+    } else {
+      # otherwise 10% of the markers are randomly sampled
+      sampled.markers <- unique(input$MARKERS)
+      sampled.markers <- sample(x = sampled.markers,
+                                size = length(sampled.markers) * 0.10,
+                                replace = FALSE)
+    }
 
```

Comment:
This is to solve the problem that with small datasets, it was possible to sample only one or two loci that weren't representative of the entire set. This was happening with my small example. I suppose this is an edge case.